### DISCUSSION TITLE
How to ignore accents ?

### DISCUSSION THREAD

**Discussion by barcell1** [ibhagwan/fzf-lua#1496] [Q&A] [answered]
like atenção = atençao . 
Is there a way to make accents be ignored (like if i search atençao would return results from atençao and atenção) ?

Thanks in advance

**Comment by ibhagwan** [comment]
That would be something handled by fzf (the binary) when fuzzy matching or a regex if using live grep, I’m unsure about the answer to either of these but both of these would be controlled upstream (by either `fzf` or `rg`).

**Reply by barcell1** [reply]
is possible somehow for me to create a file to parse to the plugin, something like this ? 

[equivalence]
a = aá
e = eé
i = ií
o = oó
u = uúü
n = nñ


I dont found a straight solution with rg

**Reply by ibhagwan** [reply]
Once the interface is open fzf (the binary) controls how characters are interpreted for fuzzy matching, fzf-lua has no control over the fuzzy matching algorithm, I doubt this is possible, try asking in the upstream repo: https://github.com/junegunn/fzf

**Reply by barcell1** [reply]
I miss more in the live grep, can you think any solution that could solve this with rg ? i dint found any , but you are more intelligent than me  

Second question, in obsidian notetaking app has a plugin called omnisearch, it caches the directory and normalize it so it can ignore accent, im wondering why fuzzy searches in neovim not choose this path of caching the directory , would this not improve speed ?

**Reply by ibhagwan** [reply, marked as answer]
> I miss more in the live grep, can you think any solution that could solve this with rg ? i dint found any , but you are more intelligent than me

Perhaps an fzf `change` event that modifies the query info a regex that contains all accents, i.e. if I press `a`, fzf will trigger the event and change it to `[aaa]` (a’s with different accents), you’ll probably need a `transform` action to achieve that, consult `man fzf` and the upstream issue tracker how to achieve that.

 > Second question, in obsidian notetaking app has a plugin called omnisearch, it caches the directory and normalize it so it can ignore accent, im wondering why fuzzy searches in neovim not choose this path of caching the directory , would this not improve speed ?

Neovim has nothing to do with this, fzf-lua uses `rg` command to search the contents of the files, you can perhaps specify `{ cwd = … }` to search another directory (cache directory) but unless the files are in the exact same format you’d need a shell command that knows how to translate the cache to it’s corresponding files.

Use `debug = true` in the option to see what underlying shell command is run, read its manual to see how to customize it and use `{ cmd = … }` to feed fzf-lua files/grep with a custom command.
